packages/beacon-node/src/chain/chain.ts

```diff
@@ -638,135 +637,105 @@ export class BeaconChain implements IBeaconChain {
-      slot,
-      {dontTransferCache: true},
-      RegenCaller.produceBlock
-    );
-
+  async produceCommonBlockBody(
+    blockAttributes: BlockAttributes & {currentState: CachedBeaconStateAllForks}
+  ): Promise<CommonBlockBody> {
     // TODO: To avoid breaking changes for metric define this attribute
     const blockType = BlockType.Full;
 
-    return produceCommonBlockBody.call(this, blockType, state, {
+    return produceCommonBlockBody.call(this, blockType, blockAttributes.currentState, {
       ...blockAttributes,
-      parentSlot: slot - 1,
+      parentSlot: blockAttributes.slot - 1,
     });
   }
 
-  produceBlock(blockAttributes: BlockAttributes & {commonBlockBody?: CommonBlockBody}): Promise<{
-    block: BeaconBlock;
-    executionPayloadValue: Wei;
-    consensusBlockValue: Wei;
-    shouldOverrideBuilder?: boolean;
-  }> {
-    return this.produceBlockWrapper<BlockType.Full>(BlockType.Full, blockAttributes);
+  async produceFullBlockBody(
+    blockAttributes: BlockAttributes & {currentState: CachedBeaconStateAllForks}
+  ): Promise<AssembledBlockBodyResponse<BlockType.Full>> {
+    const {slot, currentState} = blockAttributes;
+    const proposerIndex = currentState.epochCtx.getBeaconProposer(slot);
+    const proposerPubKey = currentState.epochCtx.index2pubkey[proposerIndex].toBytes();
+
+    return produceFullBlockBody.call(this, currentState, {
+      ...blockAttributes,
+      parentSlot: slot - 1,
+      proposerIndex,
+      proposerPubKey,
+    });
   }
 
-  produceBlindedBlock(blockAttributes: BlockAttributes & {commonBlockBody?: CommonBlockBody}): Promise<{
-    block: BlindedBeaconBlock;
-    executionPayloadValue: Wei;
-    consensusBlockValue: Wei;
-  }> {
-    return this.produceBlockWrapper<BlockType.Blinded>(BlockType.Blinded, blockAttributes);
+  async produceBlindedBlockBody(
+    blockAttributes: BlockAttributes & {currentState: CachedBeaconStateAllForks}
+  ): Promise<AssembledBlockBodyResponse<BlockType.Blinded>> {
+    const {currentState, slot} = blockAttributes;
+    const proposerIndex = currentState.epochCtx.getBeaconProposer(slot);
+    const proposerPubKey = currentState.epochCtx.index2pubkey[proposerIndex].toBytes();
+
+    return produceBlindedBlockBody.call(this, currentState, {
+      ...blockAttributes,
+      parentSlot: slot - 1,
+      proposerIndex,
+      proposerPubKey,
+    });
   }
 
-  async produceBlockWrapper<T extends BlockType>(
-    blockType: T,
-    {
-      randaoReveal,
-      graffiti,
-      slot,
-      feeRecipient,
-      commonBlockBody,
-      parentBlockRoot,
-    }: BlockAttributes & {commonBlockBody?: CommonBlockBody}
-  ): Promise<{
-    block: AssembledBlockType<T>;
-    executionPayloadValue: Wei;
-    consensusBlockValue: Wei;
-    shouldOverrideBuilder?: boolean;
-  }> {
-    const state = await this.regen.getBlockSlotState(
+  async produceBlock(blockAttributes: BlockAttributes): Promise<AssembledBlockResponse<BlockType.Full>> {
```

Comment:
this is only called by `produceEngineFullBlockOrContents` which is called by `produceBlockV2`
would like to remove it to simplify this work if it's not in use anymore but need confirmation from @nflaig and @g11tech

Reply:
Yes I knew it and and that's why isolated it's implementation from rest of the API endpoints. So we can remove it when needed in a separate PR not in this one.

Reply:
if it's me I'd just remove this `produceBlock()` + `produceBlindedBlock()` in this API because they're not used anymore after #7749 and it simplifies this work, I already see an incorrect implementation of it [here](https://github.com/ChainSafe/lodestar/pull/7740/files#r2073002505)
it'll also take less time for other reviewers
so @nazarhussain please reconsider it again